### PR TITLE
Account Restricted

### LINKED ISSUE #1304: Account in Restricted Mode
**Is your feature request related to a problem? Please describe.**
A user who on login receives a token indicating that their account is in arrears (contains the `Restricted` capability in the Access Token Files permission), the UI capabilities should be limited to actions that the user is allowed to take, ie.
- Allow: 
  -  Bucket Explorer: Download, Remove, Ls, Tree, File; 
  -  Bucket Manager: list, remove
- Deny: everything else

**UI Changes required:**
- A banner should be displayed notifying the user of their account status. The banner should contain a CTA that will allow the user to rectify the billing situation.
- Options that the user should be unable to take (such as upload, etc) should be disabled
- For functionality where there aren't UI elements that we can disable (such as DnD upload) we should handle the rejection in the FilesContext, and display a Toast notification to the user.

More details about the billing flows can be found here: 
https://www.notion.so/chainsafe/Design-Meeting-July-19-2021-ab26a38ca9c94ec48eee6d680fffc136
https://drive.google.com/file/d/13ujkxTJSt2HzqR-IPXAdywM4URJSu1yI/view

### PR DESCRIPTION
**Pull request by FSM1**
Closes #1304

### CODE REVIEW

File: packages/files-ui/src/Components/Elements/RestrictedModeBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      accountRestrictedNotification: {
+        position: "fixed",
+        bottom: 0,
+        backgroundColor: palette.additional["gray"][10],
+        color: palette.additional["gray"][1],
+        padding: "16px 24px",
+        marginLeft: 0,
+        width: "100vw",
+        [breakpoints.up("md")]: {
+          marginLeft: `${constants.navWidth}px`,
```
Comment (RyRy79261) [inline packages/files-ui/src/Components/Elements/RestrictedModeBanner.tsx]: Please use `left` instead when positioning non-relative elements

File: packages/files-ui/src/Components/Modules/FileBrowsers/SharedFoldersOverview.tsx [hunk 2]
```diff
@@ -47,6 +51,10 @@ const useStyles = makeStyles(
           minHeight: `calc(100vh - ${Number(constants.contentTopPadding)}px)`,
           "&.droppable": {
             borderColor: palette.additional["geekblue"][4]
+          },
+          "&.bottomBanner": {
+            minHeight: `calc(100vh - ${Number(constants.contentTopPadding) + 80}px)`,
+            paddingBottom: 80
```
Comment (RyRy79261) [inline packages/files-ui/src/Components/Modules/FileBrowsers/SharedFoldersOverview.tsx]: If this 80 is the value for the height of the banner in multiple files, please add it to the constants in the theme config

**Review by RyRy79261 — DISMISSED** [review]

File: packages/files-ui/src/Components/Layouts/AppWrapper.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ import AppHeader from "./AppHeader"
 import AppNav from "./AppNav"
 import { useThresholdKey } from "../../Contexts/ThresholdKeyContext"
 
+
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Layouts/AppWrapper.tsx]:
```suggestion
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx [hunk 4]
```diff
@@ -1102,12 +1121,16 @@ const FilesList = ({ isShared = false }: Props) => {
...
           filePath={currentPath}
         />
+        }
+        <SharingExplainerModal
+          showModal={showExplainerBeforeShare}
+          onHide={hideModal}
+        />
+      </article>
+      {accountRestricted &&
+      <RestrictedModeBanner />
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx]: indentation
```suggestion
        <RestrictedModeBanner />
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/SharedFileBrowser.tsx [hunk 5]
```diff
@@ -179,23 +180,20 @@ const SharedFileBrowser = () => {
...
+        subtitle: "Oops! You need to pay for this month to upload more content."
       })
-    } else {
-      uploadFiles(bucket, files, path)
-        .then(() => refreshContents())
-        .catch(console.error)
+      return
     }
-  }, [addToast, uploadFiles, bucket, refreshContents])
+    const flattenedFiles = await getFilesFromDataTransferItems(fileItems)
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/SharedFileBrowser.tsx:191]: bonus :D

File: packages/files-ui/src/Components/Modules/FileBrowsers/SharedFoldersOverview.tsx [hunk 6]
```diff
@@ -294,6 +306,9 @@ const SharedFolderOverview = () => {
         injectedClass={{ inner: classes.confirmDeletionDialog }}
         testId="shared-folder-deletion"
       />
+      {accountRestricted &&
+      <RestrictedModeBanner />
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/SharedFoldersOverview.tsx]: indentation
```suggestion
        <RestrictedModeBanner />
```

File: packages/files-ui/src/Contexts/FilesApiContext.tsx [hunk 7]
```diff
@@ -308,7 +315,8 @@ const FilesApiProvider = ({ apiUrl, withLocalStorage = true, children }: FilesAp
         thresholdKeyLogin,
         secureThresholdKeyAccount,
         encryptedEncryptionKey: decodedRefreshToken?.enckey,
-        isMasterPasswordSet: !!decodedRefreshToken?.mps
+        isMasterPasswordSet: !!decodedRefreshToken?.mps,
+        accountRestricted: accountRestricted
```
Comment (Tbaut) [inline packages/files-ui/src/Contexts/FilesApiContext.tsx]: we can simplify
```suggestion
        accountRestricted
```

File: packages/files-ui/src/Contexts/FilesApiContext.tsx [hunk 8]
```diff
@@ -62,6 +63,7 @@ const FilesApiProvider = ({ apiUrl, withLocalStorage = true, children }: FilesAp
   // access tokens
   const [accessToken, setAccessToken] = useState<Token | undefined>(undefined)
   const [secured, setSecured] = useState<boolean | undefined>(undefined)
+  const [accountRestricted, setAccountRestricted] = useState<boolean | undefined>(undefined)
```
Comment (Tbaut) [inline packages/files-ui/src/Contexts/FilesApiContext.tsx]: we can simplify
```suggestion
  const [accountRestricted, setAccountRestricted] = useState(false)
```

File: packages/files-ui/src/Components/Elements/RestrictedModeBanner.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+)
+
+const RestrictedModeBanner = () => {
+  const classes = useStyles()
+  const { desktop } = useThemeSwitcher()
+  const { redirect } = useHistory()
+
+  return (
+    <div className={classes.accountRestrictedNotification}>
+      <Typography variant='body2'>
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Elements/RestrictedModeBanner.tsx]: This is very tiny, I can barely read it :nerd_face: 
image

File: packages/files-ui/src/Components/Modules/FileBrowsers/CSFFileBrowser.tsx [hunk 10]
```diff
@@ -176,19 +176,27 @@ const CSFFileBrowser: React.FC<IFileBrowserModuleProps> = () => {
 
   const handleUploadOnDrop = useCallback(async (files: File[], fileItems: DataTransferItemList, path: string) => {
     if (!bucket) return
+    if (accountRestricted) {
+      addToast({
+        type:"error",
+        title: "Uploads disabled",
+        subtitle: "Oops! You need to pay for this month to upload more content."
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/CSFFileBrowser.tsx]:
```suggestion
        title: t`Uploads disabled`,
        subtitle: t`Oops! You need to pay for this month to upload more content.`
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/CSFFileBrowser.tsx [hunk 11]
```diff
@@ -176,19 +176,27 @@ const CSFFileBrowser: React.FC<IFileBrowserModuleProps> = () => {
...
-  }, [uploadFiles, bucket])
+  }, [uploadFiles, bucket, accountRestricted, addToast])
 
   const viewFolder = useCallback((cid: string) => {
     const fileSystemItem = pathContents.find(f => f.cid === cid)
     if (fileSystemItem && fileSystemItem.content_type === CONTENT_TYPES.Directory) {
       redirect(ROUTE_LINKS.Drive(getUrlSafePathWithFile(currentPath, fileSystemItem.name)))
     }
-  }, [currentPath, pathContents, redirect])
+  }, [currentPath, pathContents, redirect ])
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/CSFFileBrowser.tsx]:
```suggestion
  }, [currentPath, pathContents, redirect])
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/SharedFileBrowser.tsx [hunk 12]
```diff
@@ -179,23 +180,20 @@ const SharedFileBrowser = () => {
...
-      }
-    }
-    if (hasFolder) {
+    if (accountRestricted) {
       addToast({
-        title: t`Folder uploads are not supported currently`,
-        type: "error"
+        type:"error",
+        title: "Unable to upload",
+        subtitle: "Oops! You need to pay for this month to upload more content."
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/SharedFileBrowser.tsx]:
```suggestion
        title: t`Unable to upload`,
        subtitle: t`Oops! You need to pay for this month to upload more content.`
```

**Review by Tbaut — COMMENTED** [review]
Super nice and elegant for the disabled menus :rocket: Just a couple nits

**Review by Tbaut — APPROVED** [review]
:100:

**Comment by Tbaut** [comment]
Just the lint in not happy with small things fixable with a `yarn lint --fix`

File: packages/files-ui/src/Components/Elements/RestrictedModeBanner.tsx [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
+const useStyles = makeStyles(
+  ({ breakpoints, constants, palette }: CSFTheme) => {
+    return createStyles({
+      accountRestrictedNotification: {
+        position: "fixed",
+        bottom: 0,
+        backgroundColor: palette.additional["gray"][10],
+        color: palette.additional["gray"][1],
+        padding: "16px 24px",
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Components/Elements/RestrictedModeBanner.tsx]:
```suggestion
        padding: `${constants.generatUnit * 2}px ${constants.generatUnit * 3}px`,
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/ShareModal.tsx [hunk 14]
```diff
@@ -213,14 +215,16 @@ const ShareModal = ({ close, file, filePath }: IShareFileProps) => {
       .filter(buck => buck.type === "share" || buck.type === "csf")
       // filter out the current bucket
       .filter(buck => buck.id !== bucket?.id)
-      // all buckets where the user is reader or writer
+      // all buckets where the user is owner or writer
       .filter(buck => !!buck.writers.find((w) => w.uuid === profile.userId) || !!buck.owners.find((o) => o.uuid === profile.userId))
+      // filter out CSF and share buckets where user is an owner if their account is restricted
+      .filter(buck => !(!!accountRestricted && (buck.type === "csf" || !!buck.owners.find(o => o.uuid === profile.userId))))
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Components/Modules/FileBrowsers/ShareModal.tsx:221]: I would prefer the owner's shared folders to be disabled ! 
Maybe we can have a separate ticket for that !

**Review by tanmoyAtb — APPROVED** [review]
This is such complete work !

File: packages/files-ui/src/Components/Modules/FileBrowsers/ShareModal.tsx (hunk 14 above)
Comment (FSM1) [reply packages/files-ui/src/Components/Modules/FileBrowsers/ShareModal.tsx:221]: ahhh yeah that is probably doable, though might be a bit confusing for users why certain items are enabled while others are not.

**Pull request merged**